File: C/day060.c
/* Author: Ahmed A. M. ALOMAR
Date: February 29 - 2020
Description:
This program is an example of a program that will calculate the length of the string and print out a message about the calculation. */


#include <stdio.h>

int main(void)
{
  // Define and Initialize a string
  char first_sentence[] = "What a beautiful day is today";
  // Variable to count the string length
  unsigned int string_length = 0;

  // increment until the terminating character
  while (first_sentence[string_length] != '\0')
  ++string_length;

  // Print out the message
  printf("The string Length of the sentence \"%s\" is: %d characters.\n", first_sentence, string_length);

  return 0;
}

File: SourceCode/Python/day060.py
# ------------------------------------ #
# Author: Ahmed A. M. ALOMAR
# Date: June 9 - 2020
#
# Description:
# * Example of Recursion
# * This program will increment numbers
#   and print them using recursion
#   until it reaches 10
# ------------------------------------ #


#Example
## using normal while loop
# def count_down_from(number):
#   start = number
#   while start > 0:
#     print(start)
#     start -+ 1

## Using RECURSION
def count_from(number):

  if number >= 11:   # This is a condition so recursion is not go infinite
    return          # this return will end the function
    
  print(number)   # this is what will be printed
  count_from(number + 1)  # this is recursion


# this a function call
count_from(1)
